apps/src/bubbleChoice/BubbleChoice.tsx

```diff
@@ -0,0 +1,107 @@
+// BubbleChoice
+//
+// This is a React client for a bubble_choice level.  Note that this is
+// only used for levels that use Lab2.  For levels that don't use Lab2,
+// they will get an older-style level.
+
+import {Button} from '@code-dot-org/component-library/button';
+import {Heading4} from '@code-dot-org/component-library/typography';
+import classNames from 'classnames';
+import React, {useCallback, useEffect} from 'react';
+import {useSelector} from 'react-redux';
+
+import {navigateToLevelId} from '@cdo/apps/code-studio/progressRedux';
+import {LabState} from '@cdo/apps/lab2/lab2Redux';
+import continueOrFinishLesson from '@cdo/apps/lab2/progress/continueOrFinishLesson';
+import {
+  LabProps,
+  BubbleChoiceLevelData,
+  BubbleChoiceSublevel,
+} from '@cdo/apps/lab2/types';
+import {capitalizeFirstLetter} from '@cdo/apps/util/capitalizeFirstLetter';
+import {useAppDispatch, useAppSelector} from '@cdo/apps/util/reduxHooks';
+
+import {getCurrentLesson} from '../code-studio/progressReduxSelectors';
+import {commonI18n} from '../types/locale';
+
+import styles from './BubbleChoice.module.scss';
+
+const BubbleChoice: React.FunctionComponent<LabProps> = () => {
+  const dispatch = useAppDispatch();
+  const levelData = useSelector(
```

Comment:
can use `useAppSelector` to not have to cast every time 🙂

Reply:
Always gets me.  Thanks!